core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2484,6 +2497,259 @@ private long startOffsetDuringInitialization(long partitionDataStartOffset) thro
         }
     }
 
+    private ShareAcquiredRecords filterAbortedTransactionalAcquiredRecords(FetchPartitionData fetchPartitionData, FetchIsolation isolationLevel, ShareAcquiredRecords shareAcquiredRecords) {
+        if (isolationLevel != FetchIsolation.TXN_COMMITTED)
+            return shareAcquiredRecords;
+        // When FetchIsolation.TXN_COMMITTED is used as isolation type by the share group, we need to filter any
+        // transactions that were aborted/did not commit due to timeout.
+        List<AcquiredRecords> result = filterAbortedTransactionalRecords(fetchPartitionData.records.batches(), shareAcquiredRecords.acquiredRecords(), fetchPartitionData.abortedTransactions);
+        int acquiredCount = 0;
+        for (AcquiredRecords records : result) {
+            acquiredCount += (int) (records.lastOffset() - records.firstOffset() + 1);
+        }
+        return new ShareAcquiredRecords(result, acquiredCount);
+    }
+
+    private List<AcquiredRecords> filterAbortedTransactionalRecords(
```

Comment:
Shouldn't the method name be consistent with earlier method `filterAbortedTransactionalAcquiredRecords` as we are sending the `acquired` records list here as well. Hence earlier method can have `maybe` prefix and this one as `filterAbortedTransactionalAcquiredRecords`.